### ISSUE TITLE
Cannot write annotations for long recordings

### ISSUE THREAD

**Issue by spencerblackwood** [holgern/pyedflib#175]
I am trying to convert EEG data to edf+ using pyedflib, but I am having issues with annotations.

The recordings last one week at a sampling rate of 250 Hz. I am trying to add annotations to the file using EdfWriter.writeAnnotation(), but the annotations no longer write to the file when onset_in_seconds gets too big. For example, the annotation will appear when onset_in_seconds=50000, but not when onset_in_seconds=250000.

**Comment by skjerns** [comment]
Do you have a code example that can replicate the behaviour for us? That would make it much easier, thanks!

**Comment by spencerblackwood** [comment]
Here is an example mimicking what I am doing.
```python
# 5 channels for one week, write annotation at 4th day
fs = 250
data = np.random.normal(size=(5,7*24*60*60*fs))
ch_names = ['EEG1', 'EEG2', 'EEG3', 'EEG4', 'EEG5']

save_path = 'annotation_test.edf'
f = EdfWriter(save_path, len(ch_names))

sig_headers = highlevel.make_signal_headers(ch_names, sample_rate=250)
f.setSignalHeaders(sig_headers)
f.writeSamples(data)
f.writeAnnotation(4*24*3600, -1, "4 Days after start")
f.close()
```
When I run this code and load annotation_test.edf, there are no annotations. The same thing happens with highlevel.
```python
# Using highlevel instead
fs = 250
data = np.random.normal(size=(5,7*24*60*60*fs))
ch_names = ['EEG1', 'EEG2', 'EEG3', 'EEG4', 'EEG5']

save_path = 'annotation_test.edf'
sig_headers = highlevel.make_signal_headers(ch_names, sample_rate=250)

annotations = [[4*24*3600, -1, "4 Days after start"]]
header = highlevel.make_header()
header = header.update({'annotations':annotations})

highlevel.write_edf(save_path, data, sig_headers)
```
I tried the dev branch you linked, but it seems to persist there too.

**Comment by skjerns** [comment]
Thank's for the code. I think I found the bug:

https://github.com/holgern/pyedflib/blob/a616138f8ba31b94d77ae93eca1d127d1889ecf2/pyedflib/edfwriter.py#L863

The conversion to `astype(int)` somehow goes to `int32` not `int64`. By making the conversion implicit to `int64` it should correctly convert to the C `long long` which is later used. I've fixed it in #176

**Comment by spencerblackwood** [comment]
Yes, this works. Thank you for your help!
